DependencyInjection/Compiler/TokenStorageCompilerPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the FOSOAuthServerBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\OAuthServerBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Andras Ratz <ratz.andras86@gmail.com>
+ */
+class TokenStorageCompilerPass implements CompilerPassInterface
+{
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $definition = $container->getDefinition('fos_oauth_server.security.authentication.listener');
+
+        if ($container->hasDefinition('security.token_storage')) {
+            $tokenStorageReference = new Reference('security.token_storage');
```

Comment:
I would keep this one by default in the XML file, and have the compiler pass replace only when there is no token storage service. this way, we can just remove the compiler pass when the BC is not needed anymore